src/OpenLoco/Input/MouseInput.cpp

```diff
@@ -2116,27 +2117,30 @@ namespace OpenLoco::Input
         _rightMouseButtonStatus = status;
     }
 
-#pragma pack(push, 1)
-    struct QueuedMouseInput
-    {
-        uint32_t x;
-        uint32_t y;
-        uint32_t button;
-    };
-#pragma pack(pop)
+    // 0x00113E9E0
+    static std::queue<QueuedMouseInput> _mouseQueue;
 
     // 0x00406FEC
-    void enqueueMouseButton(int32_t button)
+    void enqueueMouseButton(const QueuedMouseInput& input)
     {
-        ((void (*)(int))0x00406FEC)(button);
+        constexpr uint32_t kMouseQueueSize = 64;
+        if (_mouseQueue.size() > kMouseQueueSize)
```

Comment:
should be `>=`, because if you call this function when _mouseQueue.size() is full, ie has 64 items, then this `if` 'fails' and the input is enqueued, making the queue 65 items

Reply:
The limit is arbitrary but yeah should probably make it match vanilla with a >=